### PR TITLE
fixed timestamp overriding keen addons

### PR DESCRIPTION
**Pull request by chethan2k5**
Fixed timestamp overriding keen addons when user specified custom timestamp

### CODE REVIEW

File: keen/client.py [hunk 1]
```diff
@@ -38,6 +38,10 @@ def to_json(self):
         event_as_dict = copy.deepcopy(self.event_body)
         if self.timestamp:
             event_as_dict["keen"] = {"timestamp": self.timestamp.isoformat()}
```
Comment (stroud109) [inline keen/client.py]: This line should be removed.

**Comment by stroud109** [comment]
Hey @chethan2k5, thanks so much for putting together this PR. Could you add this to `client_tests.py`? Once you remove line `client.py:40` all tests should pass. Thanks again! 

```python
class EventTests(BaseTestCase):

    def test_custom_addon(self):
        event = Event(
            '<project_id>',
            '<event_collection>',
            {'keen': {'addons': {'asdf': 1}}},
            timestamp=datetime.datetime.now(),
        )
        as_json = json.loads(event.to_json())

        self.assertEqual(as_json['keen']['addons']['asdf'], 1)
        self.assertTrue('timestamp' in as_json['keen'])
```

**Comment by chethan2k5** [comment]
@stroud109 Hey Stephanie, hope it helps. Updated with tests and review comments. Cheers

File: keen/tests/client_tests.py [hunk 2]
```diff
@@ -286,6 +286,21 @@ def url_escape(self, url):
             import urllib.parse
             return urllib.parse.quote(url)
 
+@patch("requests.Session.post")
```
Comment (stroud109) [inline keen/tests/client_tests.py]: Oops! We don't need this mock for `EventTests` right now. If you remove it, tests should pass.

**Comment by stroud109** [comment]
Looks like tests failed because `Event` is not defined. So sorry I missed that the first time around :disappointed:  Could you update [client_tests.py:9](https://github.com/keenlabs/KeenClient-Python/blob/master/keen/tests/client_tests.py#L9) so it reads `from keen.client import KeenClient, Event`? You can check if that does the trick by running tests in your terminal with `python setup.py test`.

**Comment by stroud109** [comment]
Hey @chethan2k5! Thanks again for working on this bug fix! Are you still interested in wrapping up this PR? I'd like to get the SDK updated soon, so if you don't have any more time to work on this issue, I may wrap it up on my end.

**Comment by chethan2k5** [comment]
Hey,

I can work on it tomorrow and sort it out had a lot of work for the last
week integrating with keen. I will wrap this up.

Regards,
Chethan

On Tue, Aug 18, 2015 at 3:56 PM, Stephanie Stroud <notifications@github.com>
wrote:

> Hey @chethan2k5 <https://github.com/chethan2k5>! Thanks again for working
> on this bug fix! Are you still interested in wrapping up this PR? I'd like
> to get the SDK updated soon, so if you don't have any more time to work on
> this issue, I may wrap it up on my end.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/keenlabs/KeenClient-Python/pull/77#issuecomment-132384336>
> .
>



-- 
Chethan Mittapalli
Babson MBA Class of 2014
chethan.us <http:/www.chethan.us>
linkedin

**Comment by stroud109** [comment]
Sweet! Please let me know if/how I can help :)

**Comment by stroud109** [comment]
:+1: Thanks so much @chethan2k5!

**Comment by chethan2k5** [comment]
@stroud109 np :)

**Pull request merged**